apps/judicial-system/web/src/routes/Court/InvestigationRequest/HearingArrangements/HearingArrangements.tsx

```diff
@@ -45,9 +46,20 @@ const HearingArrangements = () => {
 
   useEffect(() => {
     if (!workingCase && data?.case) {
-      setWorkingCase(data.case)
+      let theCase = data.case
+
+      if (!theCase.courtDate && theCase.requestedCourtDate) {
+        updateCase(
+          theCase.id,
+          parseString('courtDate', theCase.requestedCourtDate),
+        )
+
+        theCase = { ...theCase, courtDate: theCase.requestedCourtDate }
+      }
+
+      setWorkingCase(theCase)
     }
-  }, [workingCase, setWorkingCase, data])
+  }, [workingCase, setWorkingCase, data, updateCase])
```

Comment:
Why is courtDate not part of autofill in useCase like for example validToDate?

Reply:
I forgot about the autofill function 😱  Fixing...